addons/stock/models/stock_move.py

```diff
@@ -1525,7 +1525,7 @@ def _add_serial_move_line_to_vals_list(self, reserved_quant, quantity):
     def _should_bypass_reservation(self, forced_location=False):
         self.ensure_one()
         location = forced_location or self.location_id
-        return location.should_bypass_reservation() or self.product_id.type != 'product'
+        return location.should_bypass_reservation() or self.product_id.type != 'product' or (self.state in ('done', 'cancel') and not self._context.get('free_reservation', False))
```

Comment:
I would even not call the `should_bypass_reservation` in `stock_move_line:create()` in case of done state. 
For instance, in `write()` 

https://github.com/odoo/odoo/blob/f75244aed18d/addons/stock/models/stock_move_line.py#L413